pennylane/tape/qscript.py

```diff
@@ -546,25 +551,51 @@ def trainable_params(self, param_indices):
 
         self._trainable_params = sorted(set(param_indices))
 
-    def get_operation(self, idx):
+    def get_operation(self, idx, return_op_index=False):
         """Returns the trainable operation, and the corresponding operation argument
         index, for a specified trainable parameter index.
 
         Args:
             idx (int): the trainable parameter index
-
+            return_op_index (bool): the trainable parameter index
         Returns:
             tuple[.Operation, int]: tuple containing the corresponding
             operation, and an integer representing the argument index,
             for the provided trainable parameter.
         """
+        if return_op_index:
+            return self._get_operation(idx)
+        warnings.warn(
+            "The get_operation will soon be updated to also return the index of the trainable operation in the tape."
+            "If you want to switch to the new behavior, you can pass `return_op_index=True`"
+        )
+
         # get the index of the parameter in the script
         t_idx = self.trainable_params[idx]
 
         # get the info for the parameter
         info = self._par_info[t_idx]
         return info["op"], info["p_idx"]
 
+    def _get_operation(self, idx):
+        """Returns the trainable operation, the operation index and the corresponding operation argument
+        index, for a specified trainable parameter index.
+
+        Args:
+            idx (int): the trainable parameter index
+
+        Returns:
+            tuple[.Operation, int, int]: tuple containing the corresponding
+            operation, operation index and an integer representing the argument index,
+            for the provided trainable parameter.
+        """
+        # get the index of the parameter in the script
+        t_idx = self.trainable_params[idx]
+
+        # get the info for the parameter
+        info = self._par_info[t_idx]
+        return info["op"], info["op_idx"], info["p_idx"]
```

Comment:
As dictionaries are guaranteed to have a fixed order (the input order), we could do 
```suggestion
        return list(info.values())
```

Reply:
If we ever put the time into cleaning all this behaviour up, we could even have `_par_info` be a list of named tuples.